### DISCUSSION TITLE
kind question

### DISCUSSION THREAD

**Discussion by ukrolelo** [gijzelaerr/python-snap7#279] [Q&A] [answered]
Hi! 
Sorry to contacting you through this. I would like to ask is it possible to send a program through your library?
Or its only of the type of communication apart of modbus.
Its LOGO8, program with *.lsc
Thank you

**Comment by ukrolelo** [comment]
Ohh there is a tab for discussion, ups :>

**Comment by lautarodapin** [comment]
What do you mean with send a program? upload the program to the PLC?

**Reply by ukrolelo** [reply]
yes, sorry to not being more specific. Now i upload the program through Logosoft, but it would be awesome if i can automate uploading of the program from linux :o

**Reply by lautarodapin** [reply]
I never try that, there is a ``full_upload`` method, I never used it, maybe check it, but it seems that it is only for data blocks.

Why would you need to automate this process?

**Reply by ukrolelo** [reply]
I am not aware what are data blocks. Well the process of installing our program to PLC device is.
Person find the folder, open the program, pres connect,press test,press ok, then pres yes to start,then he needs to update time.
If i can make a script where it automates allof it it would be magical :> when i have time,will try to tinker a bit with full_upload

**Comment by DominikK0702** [comment, marked as answer]
you should check out the tia openeness api. with that you able to automate almost everything what a user can do with tia portal.

**Reply by ukrolelo** [reply]
Omg, god bless your soul <3 :)))))))))))) Thx
